src/main/kotlin/lotto/domain/LottoStatistics.kt

```diff
@@ -0,0 +1,21 @@
+package lotto.domain
+
+import kotlin.math.floor
+
+class LottoStatistics(private val matchesCountResult: List<Int>, private val amount: Int) {
+    val matchesResult: Map<LottoRule, Int> = initMatchesResult()
+
+    private fun initMatchesResult(): Map<LottoRule, Int> {
+        return LottoRule.values().associateWith { rule ->
+            matchesCountResult.count { matchesCount -> matchesCount == rule.matchesCount }
+        }
+    }
+
+    var rateOfReward: Double = calculateTotalPofit()
+        private set
+
+    private fun calculateTotalPofit(): Double {
+        var reward = matchesResult.map { result -> result.key.getTotalReward(result.value) }.sum().toDouble()
```

Comment:
```suggestion
        val reward = matchesResult.map { result -> result.key.getTotalReward(result.value) }.sum().toDouble()
```

변하지 않는 값에 대해서는 `val`를 써볼까요?